### PR TITLE
allow to configure the saleschannel in the product export command

### PR DESCRIPTION
**Pull request by breaker92**
Currently if the shop has more then one saleschannel, you can not choose the saleschannel for which you want to export.

### CODE REVIEW

**Comment by a-laurowski** [comment]
Hi @breaker92  and thank you for your contribution!
I'll take a look into that soon.

**Comment by breaker92** [comment]
@a-laurowski sure no problem, please set the correct access rights, currently im not able to push an new branch to your repository.

**Review by a-laurowski — COMMENTED** [review]
> @a-laurowski sure no problem, please set the correct access rights, currently im not able to push an new branch to your repository.

I've invited you with a write role

File: src/Command/ProductExportCommand.php [hunk 1]
```diff
@@ -44,31 +49,44 @@ class ProductExportCommand extends Command implements ContainerAwareInterface
...
         $this->addOption(self::UPLOAD_FEED_OPTION, 'u', InputOption::VALUE_NONE, 'Should upload after exporting');
         $this->addOption(self::PUSH_IMPORT_OPTION, 'i', InputOption::VALUE_NONE, 'Should import after uploading');
+        $this->addOption(self::SALESCHANNEL_OPTION, 's', InputOption::VALUE_OPTIONAL, 'ID of the saleschannel');
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $feedService = $this->feedFactory->create($this->channelService->getSalesChannelContext());
+        $salesChannel = null;
+        if($input->hasOption(self::SALESCHANNEL_OPTION)) {
```
[the remark this answers is not in the document]
Comment (breaker92) [reply src/Command/ProductExportCommand.php]: is solved

File: src/Command/ProductExportCommand.php [hunk 2]
```diff
@@ -44,31 +49,44 @@ class ProductExportCommand extends Command implements ContainerAwareInterface
...
+            $salesChannel = $this->channelRepository->search(
+                new Criteria([$input->getOption(self::SALESCHANNEL_OPTION)]),
+                new Context(new SystemSource())
```
[the remark this answers is not in the document]
Comment (breaker92) [reply src/Command/ProductExportCommand.php:86]: `SystemSource` is the most accured.
`ShopApiSource` , `AdminApiSource`  and `SalesChannelApisource` schould be used an API context.

`ShopApiSource` and `SalesChannelApisource` are require an saleschannel but we are only now determining this at this point.

`AdminApiSource` needs information about the admin user, which we do not have either.

**Comment by breaker92** [comment]
new PR with branch from this repository #13

**Pull request closed without merging**